src/classes/HH_HouseholdNaming_TEST.cls

```diff
@@ -347,10 +347,78 @@ public class HH_HouseholdNaming_TEST {
         system.assertequals('Test ' + system.label.npo02.HouseholdNameConnector + ' Jones ' + system.label.npo02.DefaultHouseholdName, hh.Name);
         system.assertequals('Jim ' + system.label.npo02.HouseholdNameConnector  + ' Joe Test ' + system.label.npo02.HouseholdNameConnector + ' John Jones', hh.npo02__Formal_greeting__c);
         system.assertequals('Jim, Joe ' + system.label.npo02.HouseholdNameConnector + ' John', hh.npo02__Informal_greeting__c);
-        
-        
     }  
 
+    /*********************************************************************************************************
+    @description
+        Test updateNames() updates Household names only for specified Accounts or Households. 
+        Contacts having either Account or Household null are not processed unless their 
+        Household/Account is in the updateNames() argument list.
+    verify:
+        Names for specified Account/Households are updated.
+    **********************************************************************************************************/  
+    private static testMethod void testUpdateNamesIsAppliedToSpecifiedHHsOnly() {
+        UTIL_UnitTestData_TEST.turnOffAutomaticHHNaming();
+        UTIL_UnitTestData_TEST.setupHHNamingSettings();
+        UTIL_CustomSettingsFacade.getContactsSettingsForTests(new npe01__Contacts_and_Orgs_Settings__c(
+            npe01__Account_Processor__c = CAO_Constants.HH_ACCOUNT_PROCESSOR
+        ));
+
+        List<Account> accounts = buildHHAccounts(2);
+        insert accounts;
+
+        List<npo02__Household__c> hhs = buildHouseholds(2);
+        insert hhs;
+
+        List<Contact> contacts = new List<Contact>{
+            new Contact(FirstName = 'Alpha', LastName = 'A', AccountId = accounts[0].Id, npo02__Household__c = hhs[0].Id),
```

Comment:
Are both AccountId and npo02__Household__c needed?

Reply:
The Households are not required in this test; they have been removed.